### DISCUSSION TITLE
How to prevent writing a site to history?

### DISCUSSION THREAD

**Discussion by goodwin74** [cefsharp/CefSharp#3325] [General]
For example, there is a start page "**browser:\\startpage**" (a SchemeHandlerFactory has been created for this page). From this page the user can go to other sites and return to the start page using the "Back" button.

How can I prevent this page from being written to history, so that after switching to another site it is impossible to go back?
I would be very grateful for a hint on how to implement this :)

**p.s. WinForms / VS2019 / C# / CefSharp 86**

**Comment by goodwin74** [comment]
I want to say right away that I tried to implement this through **OnBeforeBrowse**.
`if (request.Url.StartsWith("browser:") && (request.TransitionType == TransitionType.ForwardBack|| request.TransitionType == (TransitionType)50331649))
            {
                return true;
            }`

But there is a problem. You cannot go back, but the **back button** and the "**CanBack**" property are still active. Therefore, I think, is it possible to prevent the site from being recorded in history at all?

**Comment by amaitland** [comment]
`Chromium` has the concept of an `interstitial page` which it uses for displaying the start page/errors (they don't appear in the history), unfortunately `CEF` doesn't as yet support this. There was some discussion a while ago (see https://bitbucket.org/chromiumembedded/cef/issues/2586/loadstring-results-in-terminating-renderer#comment-52020179) though I haven't seen any progressing on this.

You can try using the `DevTools` api and calling [Page.resetNavigationHistory](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-resetNavigationHistory) after you navigate away from your start page.

Look something like the following (I haven't run this code so might need some minor corrections).
```c#
using (var client = chromiumWebBrowser.GetDevToolsClient())
{
  _ = client.Page.ResetNavigationHistoryAsync();
}
```

**Reply by goodwin74** [reply]
thank you!!
